sources/packages/backend/workflow/test/2023-2024/assessment-gateway/assessment-gateway.originalAssessment.e2e-spec.ts

```diff
@@ -57,4 +78,227 @@ describe(`E2E Test Workflow assessment gateway on original assessment for ${PROG
+          incomeVerificationId: incomeVerificationId++,
+          subprocesses: WorkflowSubprocesses.Parent1IncomeVerification,
+        }),
+        ...createIncomeRequestTaskMock({
+          incomeVerificationId: incomeVerificationId++,
+          subprocesses: WorkflowSubprocesses.Parent2IncomeVerification,
+        }),
+        ...createCheckIncomeRequestTaskMock({
+          subprocesses: WorkflowSubprocesses.StudentIncomeVerification,
+        }),
+        ...createCheckIncomeRequestTaskMock({
+          subprocesses: WorkflowSubprocesses.Parent1IncomeVerification,
+        }),
+        ...createCheckIncomeRequestTaskMock({
+          subprocesses: WorkflowSubprocesses.Parent2IncomeVerification,
+        }),
+      },
+      requestTimeout: PROCESS_INSTANCE_CREATE_TIMEOUT,
+    };
+
+    // Act
+    const assessmentGatewayResponse =
+      await zeebeClientProvider.createProcessInstanceWithResult(
+        assessmentStartData,
+      );
+
+    // Assert
+    expectToPassThroughServiceTasks(
+      assessmentGatewayResponse.variables,
+      WorkflowServiceTasks.AssociateWorkflowInstance,
+      WorkflowServiceTasks.VerifyApplicationExceptions,
+      WorkflowServiceTasks.ProgramInfoNotRequired,
+      WorkflowSubprocesses.StudentIncomeVerification,
+      WorkflowSubprocesses.RetrieveSupportingInfoParent1,
+      WorkflowSubprocesses.RetrieveSupportingInfoParent2,
+      WorkflowSubprocesses.Parent1IncomeVerification,
+      WorkflowSubprocesses.Parent2IncomeVerification,
+      WorkflowServiceTasks.SaveDisbursementSchedules,
+      WorkflowServiceTasks.AssociateMSFAA,
+      WorkflowServiceTasks.UpdateNOAStatusToRequired,
+    );
+    expectNotToPassThroughServiceTasks(
+      assessmentGatewayResponse.variables,
+      WorkflowSubprocesses.PartnerIncomeVerification,
+    );
+  });
+
+  it("Should check for parent 1 income when the student is dependant and informed that he has only one parent with SIN.", async () => {
+    // Arrange
+    const currentAssessmentId = assessmentId++;
+    const parent1SupportingUserId = supportingUserId++;
+    // Assessment consolidated mocked data.
+    const assessmentConsolidatedData: AssessmentConsolidatedData = {
+      assessmentTriggerType: AssessmentTriggerType.OriginalAssessment,
+      ...createFakeConsolidatedFulltimeData(PROGRAM_YEAR),
+      ...createParentsData({ numberOfParents: 1 }),
+      // Application with PIR not required.
+      studentDataSelectedOffering: 1,
+    };
+
+    const assessmentStartData = {
+      bpmnProcessId: "assessment-gateway",
+      variables: {
+        [ASSESSMENT_ID]: currentAssessmentId,
+        ...createLoadAssessmentConsolidatedDataMock({
+          assessmentConsolidatedData,
+        }),
+        ...createVerifyApplicationExceptionsTaskMock(),
+        ...createCreateSupportingUsersParentsTaskMock({
+          supportingUserIds: [parent1SupportingUserId],
+        }),
+        ...createCheckSupportingUserResponseTaskMock({
```

Comment:
mm, for some we are using the function name as `<task name>TaskMock` (eg, line 198, etc..), and for some, it `create<task name>TaskMock` (eg, line 193). maybe we can stick to one. let me know if I am missing anything

Reply:
Actually, I don't have a strong feeling for any one of them. I am okay with both, I was saying can we stick to any one format.Maybe the first option?